test/unit/test_result_batch.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2012-2021 Snowflake Computing Inc. All right reserved.
+#
+
+import importlib
+from collections import namedtuple
+from http import HTTPStatus
+from unittest import mock
+
+import pytest
+from mock import Mock
+
+from snowflake.connector import DatabaseError, InterfaceError
+from snowflake.connector.compat import (
+    BAD_GATEWAY,
+    BAD_REQUEST,
+    FORBIDDEN,
+    GATEWAY_TIMEOUT,
+    INTERNAL_SERVER_ERROR,
+    METHOD_NOT_ALLOWED,
+    OK,
+    REQUEST_TIMEOUT,
+    SERVICE_UNAVAILABLE,
+    UNAUTHORIZED,
+)
+from snowflake.connector.errorcode import (
+    ER_FAILED_TO_CONNECT_TO_DB,
+    ER_FAILED_TO_REQUEST,
+)
+from snowflake.connector.errors import (
+    BadGatewayError,
+    BadRequest,
+    ForbiddenError,
+    GatewayTimeoutError,
+    InternalServerError,
+    MethodNotAllowed,
+    OtherHTTPRetryableError,
+    ServiceUnavailableError,
+)
+from snowflake.connector.result_batch import MAX_DOWNLOAD_RETRY, JSONResultBatch
+from snowflake.connector.sqlstate import (
+    SQLSTATE_CONNECTION_REJECTED,
+    SQLSTATE_CONNECTION_WAS_NOT_ESTABLISHED,
+)
+
+REQUEST_MODULE_PATH = (
+    "snowflake.connector.vendored.requests"
+    if importlib.util.find_spec("snowflake.connector.vendored.requests")
+    else "requests"
+)
```

Comment:
We usually do something similar to:
```python
try:  # pragma: no cover
	from snowflake.connector.vendored import requests
    # Set a variable here for path
except ImportError:
	import requests
	# Set same variable to other path
```
I wonder if you see any upside to tapping into `importlib` instead. If so, do you think that we could turn this into a function and use it everywhere where we do the fall back imports for old driver tests?

Reply:
In this case I actually don't need to import, I only needed a way to check which dependency was available to set `REQUEST_MODULE_PATH` accordingly. I initially had what you suggested but this would result in "unused import" errors by the linting check so I decided to do this and avoid the import entirely.

This might be beneficial in cases where we don't want to do the import (like here). Some people also suggest not doing the try..except way in case the module we try importing actually does exist but one of ITS imports raises the ImportError. 
I was following the recommendation here: https://stackoverflow.com/a/14050282
I can't think of any other reasons other than readability(?) and maybe some flexibility down the line (since the importlib module seems to have some other useful functions)

Reply:
If you just want to silence the unused import statement (or other) warnings you can always put `# NOQA` after a line

Reply:
Oh I see, thanks, thats good to know in the future. I think in this case though I prefer doing it with `importlib` just to avoid importing since all I need is to set this variable.